bluepysnap/config.py

```diff
@@ -163,11 +164,54 @@ def to_dict(self):
 class CircuitConfig(Config):
     """Handle CircuitConfig."""
 
+    def __init__(self, *args):
+        """Initializes circuit config."""
+        super().__init__(*args)
+        self._populations = None
+        self._resolve_population_configs()
+
     @classmethod
     def from_config(cls, config_path):
         """Instantiate the config class from circuit configuration."""
         return cls(config_path, libsonata.CircuitConfig)
 
+    @property
+    def populations(self):
+        """Access population configs."""
+        return self._populations
+
+    # NOTE: These are parsed here to keep Parser returning the exact parsed configuration.
+    def _resolve_population_configs(self):
+        """Resolves population configs for the node and edge populations."""
+        # All the "or"s exist to have empty dicts lists instead of None's to overcome issues like:
+        # components: null in the config
+        config = self.to_dict()
+        networks = config.get("networks") or {}
+        components = config.get("components") or {}
+
+        def resolve_network_populations(element_type):
+            populations = {}
+            type_file_key = f"{element_type}_file"
+            element_dicts = networks.get(element_type) or []
+
+            for elem_dict in element_dicts:
+                elem_pops = elem_dict.get("populations") or {}
+
+                for pop, cfg in elem_pops.items():
+                    populations[pop] = copy.deepcopy(components)
+                    populations[pop].update(cfg or {})
```

Comment:
According to the specs I don't think that `cfg` can be None, so the `or` shouldn't be needed.

Reply:
Yeah, so, did I. However, the spec does not forbid a config with something like:
```json
...
"populations": {
    "default": null
}
```
In which case this would happen and, as a matter of fact, that is what happens in tests.

Reply:
About dumping to json, it's weird 🤔 because
```python
json.dumps({"populations": {"default": {}}})
```
gives
```
'{"populations": {"default": {}}}'
```
so there should be something in the middle doing some transformation.

However, keeping the `or` fallback doesn't hurt 👍

Reply:
I rechecked the test. I misremembered. It wasn't the `json.dumps` that would cause the issue, it would be `libsonata.CircuitConfig.expanded_json` (or something that creates the output) that causes the coercion from `{}` to `null`.